src/Components/WebAssembly/Server/src/DebugProxyLauncher.cs

```diff
@@ -0,0 +1,119 @@
+            var executablePath = LocateDebugProxyExecutable(environment);
+            var muxerPath = DotNetMuxer.MuxerPathOrDefault();
+            var ownerPid = Process.GetCurrentProcess().Id;
+            var processStartInfo = new ProcessStartInfo
+            {
+                FileName = muxerPath,
+                Arguments = $"exec \"{executablePath}\" --owner-pid {ownerPid}",
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+            };
+            RemoveUnwantedEnvironmentVariables(processStartInfo.Environment);
+
+            var debugProxyProcess = Process.Start(processStartInfo);
+            var nowListeningRegex = new Regex("^\\s*Now listening on: (.*)$");
+            debugProxyProcess.OutputDataReceived += (sender, eventArgs) =>
+            {
+                var match = nowListeningRegex.Match(eventArgs.Data);
+                if (match.Success)
+                {
+                    var url = match.Groups[1].Captures[0].Value;
+                    tcs.TrySetResult(url);
+                }
+            };
+            debugProxyProcess.BeginOutputReadLine();
+
+            _ = Task.Run(async () =>
+            {
+                await Task.Delay(DebugProxyLaunchTimeout);
+                tcs.TrySetException(new TimeoutException($"Failed to start the debug proxy within the timeout period of {DebugProxyLaunchTimeout.TotalSeconds} seconds."));
+            });
+
+            return await tcs.Task;
+        }
+
+        private static void RemoveUnwantedEnvironmentVariables(IDictionary<string, string> environment)
+        {
+            // Generally we expect to pass through most environment variables, since dotnet might
+            // need them for arbitrary reasons to function correctly. However, we specifically don't
+            // want to pass through any ASP.NET Core hosting related ones, since the child process
+            // shouldn't be trying to use the same port numbers, etc. In particular we need to break
+            // the association with IISExpress and the MS-ASPNETCORE-TOKEN check.
+            var keysToRemove = environment.Keys.Where(key => key.StartsWith("ASPNETCORE_")).ToList();
+            foreach (var key in keysToRemove)
+            {
+                environment.Remove(key);
+            }
+        }
+
+        private static string LocateDebugProxyExecutable(IWebHostEnvironment environment)
+        {
+            var assembly = Assembly.Load(environment.ApplicationName);
+            var assemblyLocation = GetAssemblyLocation(assembly);
+            var debugProxyPath = Path.Combine(
+                Path.GetDirectoryName(assemblyLocation),
+                "BlazorDebugProxy",
+                "Microsoft.AspNetCore.Components.WebAssembly.DebugProxy.dll");
+
+            if (!File.Exists(debugProxyPath))
+            {
+                throw new InvalidOperationException(
```

Comment:
```suggestion
                throw new FileNotFoundException(
```

Reply:
Also, how does this happen? Packaging issue?

Reply:
It would only happen if someone is manually modifying their build/publish output. In other words, it shouldn't happen, but if it does, it's nice to be told about it.